### PR TITLE
Add CI configuration and fix CI warnings

### PR DESCRIPTION
**Pull request by hky1999**

### CODE REVIEW

**Comment by aarkegz** [comment]
This pr needs a better name.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+on: [push, pull_request]
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        rust-toolchain: [nightly]
+        targets: [x86_64-unknown-linux-gnu, x86_64-unknown-none, riscv64gc-unknown-none-elf, aarch64-unknown-none-softfloat]
```
Comment (aarkegz) [inline .github/workflows/ci.yml:12]: Maybe move `x86_64-unknown-linux-gnu` to a new `job`? It's only required for unit tests.

**Review by aarkegz — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (hky1999) [reply .github/workflows/ci.yml:12]: It's copied from [arceos-crates/templates](https://github.com/arceos-org/arceos-crates/tree/main/templates).

Comment (hky1999) [reply .github/workflows/ci.yml:12]: Considering this is an OS-independent crate, this `x86_64-unknown-linux-gnu` target is to make the checks for the crate more general? @equation314

Comment (equation314) [reply .github/workflows/ci.yml:12]: To build and run unit tests on host Linux (for cargo clippy & cargo test).

Comment (equation314) [reply .github/workflows/ci.yml:12]: In `arceos-crates`, some crates may have special considerations for `target_os = "linux"`.

Comment (aarkegz) [reply .github/workflows/ci.yml:12]: OK

Comment (hky1999) [reply .github/workflows/ci.yml:12]: > OK

need ning's approvement

**Review by aarkegz — APPROVED** [review]

**Pull request merged**